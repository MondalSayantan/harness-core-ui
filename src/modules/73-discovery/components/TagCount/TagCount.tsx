/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { Icon, IconName, Layout, Tag, Text } from '@harness/uicore'
import { Color, Intent } from '@harness/design-system'
import React from 'react'
import { PopoverInteractionKind, Tooltip } from '@blueprintjs/core'
import { cloneDeep } from 'lodash-es'
import css from './TagCount.module.scss'

export interface TagCountProps {
  tagItems: string[]
  icon?: IconName
  tagCount: number
  tooltipHeader: string
}
export default function TagCount({ tagItems, icon, tagCount, tooltipHeader }: TagCountProps): React.ReactElement {
  const deepClonedItems = cloneDeep(tagItems)
  const splicedItems = deepClonedItems.splice(0, tagCount)
  return (
    <Layout.Horizontal>
      {splicedItems.map(item => {
        return (
          <div key={item} className={css.tag}>
            <Tag intent={Intent.PRIMARY}>{item}</Tag>
          </div>
        )
      })}
      {deepClonedItems.length > 0 && (
        <Tooltip
          interactionKind={PopoverInteractionKind.HOVER}
          hoverCloseDelay={500}
          content={
            <Layout.Vertical spacing={'small'} style={{ padding: 10 }}>
              <Text color={Color.WHITE} font={{ weight: 'bold' }}>
                {tooltipHeader}
              </Text>
              {deepClonedItems.map(item => {
                return (
                  <div key={item} style={{ display: 'flex', alignItems: 'center' }}>
                    {icon && (
                      <div className={css.iconDiv}>
                        <Icon name={icon} size={14} className={css.moduleIcons} />
                      </div>
                    )}
                    <Text color={Color.WHITE} margin={{ left: 'small' }}>
                      {item}
                    </Text>
                  </div>
                )
              })}
            </Layout.Vertical>
          }
        >
          <Tag intent={Intent.NONE} style={{ marginRight: 10 }}>
            {`+${deepClonedItems.length}`}
          </Tag>
        </Tooltip>
      )}
    </Layout.Horizontal>
  )
}
