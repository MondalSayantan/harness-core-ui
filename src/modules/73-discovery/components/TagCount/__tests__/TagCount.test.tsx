/*
 * Copyright 2023 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render } from '@testing-library/react'
import TagCount, { TagCountProps } from '../TagCount'

const mockData: TagCountProps = {
  tagItems: ['Test1', 'Test2', 'Test3', 'Test4', 'Test5'],
  icon: 'code-settings',
  tagCount: 2,
  tooltipHeader: 'Test headers'
}

describe('TagCount component test', () => {
  test('should match snapshot with mock data', async () => {
    const { container } = render(<TagCount {...mockData} />)
    expect(container).toMatchSnapshot()
  })
  test('validate tooltip items count', () => {
    const { container } = render(<TagCount {...mockData} />)
    expect(container).toHaveTextContent('+3')
  })
})
