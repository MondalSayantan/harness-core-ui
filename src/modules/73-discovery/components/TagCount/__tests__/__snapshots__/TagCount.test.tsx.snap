// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagCount component test should match snapshot with mock data 1`] = `
<div>
  <div
    class="Layout--horizontal StyledProps--main"
  >
    <div
      class="tag"
    >
      <span
        class="Tag--main"
      >
        <span
          class="bp3-tag bp3-intent-primary"
        >
          <span
            class="bp3-text-overflow-ellipsis bp3-fill"
          >
            Test1
          </span>
        </span>
      </span>
    </div>
    <div
      class="tag"
    >
      <span
        class="Tag--main"
      >
        <span
          class="bp3-tag bp3-intent-primary"
        >
          <span
            class="bp3-text-overflow-ellipsis bp3-fill"
          >
            Test2
          </span>
        </span>
      </span>
    </div>
    <span
      class="bp3-popover-wrapper"
    >
      <span
        class="bp3-popover-target"
      >
        <span
          class="Tag--main"
        >
          <span
            class="bp3-tag"
            style="margin-right: 10px;"
          >
            <span
              class="bp3-text-overflow-ellipsis bp3-fill"
            >
              +3
            </span>
          </span>
        </span>
      </span>
    </span>
  </div>
</div>
`;
